dali/kernels/signal/decibel/decibel_calculator.h

```diff
@@ -35,19 +36,25 @@ struct DecibelCalculator {
     assert(inv_s_ref_ > 0.0);
   }
 
-  DALI_FORCEINLINE
+  DALI_HOST_DEV DALI_FORCEINLINE
   T operator()(T s) const {
     T ratio = s * inv_s_ref_;
-    return mul_log2_ * std::log2(std::max(min_ratio_, ratio));
+#ifndef __CUDA_ARCH__
+    using std::log2;
+#endif
+    return mul_log2_ * log2(cuda_max(min_ratio_, ratio));
   }
 
-  DALI_FORCEINLINE
+  DALI_HOST_DEV DALI_FORCEINLINE
   T db2signal(T db) const {
-    return s_ref_ * std::pow(10.0, db * inv_mul_);
+#ifndef __CUDA_ARCH__
+    using std::pow;
+#endif
+    return s_ref_ * pow(10.0, db * inv_mul_);
```

Comment:
That's using double precision. CUDA doesn't have overloads for math functions, AFAIK - we should explicitly use pow/powf. That was one of the reasons I didn't use DecibelCalculator in my FFT posprocessing.